### PR TITLE
Extract various extraction toggles

### PR DESCRIPTION
**Pull request by einarmo**
Makes for more readable code, and improves the distinction between FDM (in the pusher) and loading from OPC-UA, even if the two are somewhat related.

### CODE REVIEW

File: Extractor/Config/ConfigToggles.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        /// <summary>
+        /// Should the extractor get references to types when loading non-hierarchical references?
+        /// </summary>
+        public bool GetNonHierarchicalTypeReferences => FdmEnabled || config.Extraction.NodeTypes.AsNodes;
+        /// <summary>
+        /// Should the extractor create unmapped nodes it encounters when reading non-hierarchical references, or just ignore them?
+        /// </summary>
+        public bool CreateUnknownReferencedNodes => FdmEnabled || config.Extraction.Relationships.CreateReferencedNodes;
+        /// <summary>
+        /// Should the extractor create nodes with NodeClass ObjectType, ReferenceType, VariableType, or DataType?
```
Comment (ozangoktan) [inline Extractor/Config/ConfigToggles.cs:30]: This sounds like the bool makes a distinction between 4 different cases, can we improve the comment here?

File: Extractor/Config/ConfigToggles.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        // Type manager
+        /// <summary>
+        /// Should the type manager load data types when reading the type hierarchy?
+        /// </summary>
+        public bool LoadDataTypes => FdmEnabled || config.Extraction.DataTypes.AutoIdentifyTypes;
+        /// <summary>
+        /// Should the type manager load event types when reading the type hierarchy?
+        /// Since event types are also type definitions, this is enabled by default if
+        /// LoadTypeDefinitions is true.
```
Comment (ozangoktan) [inline Extractor/Config/ConfigToggles.cs:42]: This refers to some external logic. Perhaps we should exclude this information from here

File: Extractor/Config/ConfigToggles.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        /// </summary>
+        public bool LoadEventTypes => config.Events.Enabled;
+        /// <summary>
+        /// Should the type manager load reference types when reading the type hierarchy?
+        /// </summary>
+        public bool LoadReferenceTypes => FdmEnabled || config.Extraction.Relationships.Enabled;
+        /// <summary>
+        /// Should the type manager load object- and variable types when reading the type hierarchy?
```
Comment (ozangoktan) [inline Extractor/Config/ConfigToggles.cs:50]:
```suggestion
        /// Should the type manager load object and variable types when reading the type hierarchy?
```

File: Extractor/Config/ConfigToggles.cs (hunk 2 above)
Comment (einarmo) [reply Extractor/Config/ConfigToggles.cs:42]: Sure

File: Extractor/Config/ConfigToggles.cs (hunk 1 above)
Comment (einarmo) [reply Extractor/Config/ConfigToggles.cs:30]: They are toggles, each enable a feature independently.

File: Extractor/Config/ConfigToggles.cs (hunk 2 above)
Comment (einarmo) [reply Extractor/Config/ConfigToggles.cs:42]: Actually, this doesn't refer to external logic. It refers to the structure of the base OPC-UA type hierarchy, so it is inherent to OPC-UA.

The type definition hierarchy looks like this:

 - ObjectTypes
   - EventTypes
 - VariableTypes

so if type definitions are enabled, we will get event types as well.

File: Extractor/Config/ConfigToggles.cs (hunk 1 above)
Comment (einarmo) [reply Extractor/Config/ConfigToggles.cs]: I improved the docs a bit

**Review by ozangoktan — APPROVED** [review]
LGTM 🔥

**Pull request merged**